tests/functional/backend/corpora/test_api.py

```diff
@@ -199,23 +199,24 @@ def test_dataset_upload_flow(self):
 
         with self.subTest("Test dataset conversion"):
             keep_trying = True
-            upload_statuses = ["WAITING", "UPLOADING", "UPLOADED"]
-            conversion_statuses = ["CONVERTING", "CONVERTED", "FAILED"]
+            expected_upload_statuses = ["WAITING", "UPLOADING", "UPLOADED"]
+            # conversion statuses can be `None` when/if we hit the status endpoint too early after an upload
+            expected_conversion_statuses = ["CONVERTING", "CONVERTED", "FAILED", None]
             timer = time.time()
             while keep_trying:
                 data = None
                 res = requests.get(f"{self.api}/dp/v1/datasets/{dataset_uuid}/status", headers=headers)
                 res.raise_for_status()
                 data = json.loads(res.content)
                 upload_status = data["upload_status"]
-                self.assertIn(upload_status, upload_statuses)
+                self.assertIn(upload_status, expected_upload_statuses)
```

Comment:
I would roll back this and do:
```if upload_status: 
    self.assertIn(upload_status)
```

Reply:
Ah, right, we _could_ hit `status` before an upload status is even available.

Reply:
Fixed